src/app/rundown-view/factories/key-binding.factory.ts

```diff
@@ -108,8 +110,13 @@ export class KeyBindingFactory {
   }
 
   public createRundownKeyBindings(rundown: Rundown): StyledKeyBinding[] {
+    let keybindings: StyledKeyBinding[] = [
```

Comment:
The complexity of this method is slowly increasing due to the overwrites of values and conditional updates. I suggest that the creation of key bindings are split up into groups which are based on the rundown state. My suggestion is to have the creation of rundown mode transitions key bindings (activation, entering rehearsal and deactivation) and the rundown interaction key bindings (reset, take, set next, mini shelf cycle) into their own categories and then use composition to chain them together. I have provided a suggestion below with this categorization.

Sidenote: It also looks like that entering rehearsal have no key binding yet, so I have added it in the suggestion with the key binding `Control + Backquote`.

```typescript
class KeyBindingFactory {
  public createRundownKeyBindings(rundown: Rundown): StyledKeyBinding[] {
    return [
      ...this.createRundownModeTransitionKeyBindings(rundown),
      ...this.createRundownInteractionKeyBindings(rundown),
    ]
  }

  private createRundownModeTransitionKeyBindings(rundown: Rundown): StyledKeyBinding[] {
    switch (rundown.mode) {
      case RundownMode.ACTIVE:
        return [
          this.createRundownKeyBinding('Deactivate Rundown', ['Control', 'Shift', 'Backquote'], () => this.deactivateRundown(rundown)),
        ]
      case RundownMode.REHEARSAL:
        return [
          this.createRundownKeyBinding('Activate Rundown', ['Backquote'], () => this.dialogConfirmationService.switchActivateRundownDialog(rundown)),
          this.createRundownKeyBinding('Deactivate Rundown', ['Control', 'Shift', 'Backquote'], () => this.deactivateRundown(rundown)),
        ]
      case RundownMode.INACTIVE:
        return [
          this.createRundownKeyBinding('Activate Rundown', ['Backquote'], () => this.dialogConfirmationService.switchActivateRundownDialog(rundown)),
          this.createRundownKeyBinding('Enter Rehearsal', ['Control', 'Backquote'], () => this.dialogConfirmationService.switchRehearsalRundownDialog(rundown)),
        ]
    }
  }

  private createRundownInteractionKeyBindings(rundown: Rundown): StyledKeyBinding[] {
    switch (rundown.mode) {
      case RundownMode.ACTIVE:
      case RundownMode.REHEARSAL:
        return [
          this.createRundownKeyBinding('Take', ['AnyEnter'], () => this.takeNext(rundown)),
          this.createRundownKeyBinding('Reset Rundown', ['Escape'], () => this.resetRundown(rundown)),
          this.createRundownKeyBinding('Set Segment Above as Next', ['Shift', 'ArrowUp'], () => this.setSegmentAboveNextAsNext(rundown)),
          this.createRundownKeyBinding('Set Segment Below as Next', ['Shift', 'ArrowDown'], () => this.setSegmentBelowNextAsNext(rundown)),
          this.createRundownKeyBinding('Set Earlier Part as Next', ['Shift', 'ArrowLeft'], () => this.setEarlierPartAsNext(rundown)),
          this.createRundownKeyBinding('Set Later Part as Next', ['Shift', 'ArrowRight'], () => this.setLaterPartAsNext(rundown)),
          this.createRundownKeyBinding('Cycle MiniShelf', ['Tab'], () => this.miniShelfCycleService.cycleMiniShelfForward(rundown)),
          this.createRundownKeyBinding('Cycle MiniShelf', ['Shift', 'Tab'], () => this.miniShelfCycleService.cycleMiniShelfBackward(rundown)),
        ]
      case RundownMode.INACTIVE:
        return [
          this.createRundownKeyBinding('Reset Rundown', ['Escape'], () => this.resetRundown(rundown)),
        ]
    }
  }
}
```
In the suggestion I have used a switch statement to differentiate between what system actions are available for the given rundown modes. It makes it easy to extend and have a low complexity.

Reply:
This method has been change in https://github.com/tv2/ng-sofie/pull/142
I highly suggest to not start refactoring it in this PR

Reply:
> 
> Sidenote: It also looks like that entering rehearsal have no key binding yet, so I have added it in the suggestion with the key binding `Control + Backquote`.
> 
I was on this same page and introduced (a different 'Shift+Backquote' in resemblance of the MiniShelf cycling in 'opposite' ) key binding and later removed it here https://github.com/tv2/ng-sofie/pull/131/commits/f0338066955a749ebc0e33c5c998ca86751f6fd9 
So at the end this appears to be a really missing feature.